src/core/lang.c

```diff
@@ -127,6 +128,21 @@ static void parse_message(buffer *buf)
     buffer_read_raw(buf, &data.message_data, MAX_MESSAGE_DATA);
 }
 
+static void set_event_message_parameters(lang_message* m, int title, int text, int urgent) {
```

Comment:
I am not sure about this duplication, however I am not sure about adding new parameter to `set_message_parameters` either. Both has pros and cons. Another approach might be checking for `m->message_type` to be set and default to `MESSAGE_TYPE_GENERAL`.

Reply:
Since this is a static function, I think a better approach would be to use the same function but add an extra param, `int message_type`, then the function is called with either `MESSAGE_TYPE_GENERAL` or `MESSAGE_TYPE_EVENT` as a param.

In general, avoid code duplication, prefer adding parameters to existing functions (within reason).